File: C++/OlympicAthletics/Athlete.cpp
#include "Athlete.h"

using namespace std;

//constructors:
Athlete::Athlete() {              //s predefinisanim vrednostima
        gender = NOT_SPECIFIED;
        fullName = " ";
        discipline = NO_DISCIPLINE;
        double qualificationScore = 0.00;
}
Athlete::Athlete(Gender gender, string fullName, AthleticsDiscipline discipline, double qualificationScore) { //s parametrima
		this->gender = gender;
		this->fullName = fullName;
		this->discipline = discipline;
		this->qualificationScore = qualificationScore;
}
Athlete::Athlete (const Athlete &A){ //konstruktor kopije
        gender = A.gender;
        fullName = A.fullName;
        discipline = A.discipline;
        qualificationScore = A.qualificationScore;
}

//getteri
Gender Athlete::getGender() const {return gender;}
string Athlete::getFullName() const {return fullName;}
AthleticsDiscipline Athlete::getDiscipline() const {return discipline;}
double Athlete::getQualificationScore() const {return qualificationScore;}

//seteri
void Athlete::setGender (Gender sex) {gender = sex;}
void Athlete::setFullName (string name) {fullName = name;}
void Athlete::setDiscipline (AthleticsDiscipline sport) {discipline = sport;}
void Athlete::setQualificationScore(double score) {qualificationScore = score;}


//operator ispisa za atletičare
//inline
ostream& operator<<(ostream& osA, const Athlete& atl){
    osA << atl.getFullName() << ", compete " << athleticsDiscipline[atl.discipline] << " with qualification score: " << to_string(atl.qualificationScore);
    return osA;
}

File: javaTest/Athlethics/src/competition/Athlete.java
package competition;

import sport.AthleticsDiscipline;

public class Athlete {

	private Gender gender;
	private String fullName;
	private AthleticsDiscipline discipline;
	private double qualificationScore;

	public Athlete(Gender gender, String fullName) {
		this.gender = gender;
		this.fullName = fullName;
	}

	public Gender getGender() {
		return this.gender;
	}

	public String getFullName() {
		return fullName;
	}

	public void setDiscipline(AthleticsDiscipline sport) {
		this.discipline = sport;
	}

	public AthleticsDiscipline getDiscipline() {
		return this.discipline;
	}

	public double getQualificationScore() {
		return qualificationScore;
	}

	public void setQualificationScore(double score) {
		this.qualificationScore = score;
	}

	@Override
	public String toString() {
		return new StringBuilder().append(fullName).append(", compete ").append(discipline).append(" with qualification score: ").append(qualificationScore)
				.toString();
	}
}
